tests/framework/components/test_component.py

```diff
@@ -340,3 +342,151 @@ def test_listeners_are_registered_at_custom_priorities():
     assert component.on_simulation_end in set(simulation_end_methods.get(1, []))
+
+
+def test_component_lookup_table_configuration(hdf_file_path):
+    # Tests that lookup tables are created correctly based on their configuration
+
+    favorite_team = pd.DataFrame(
+        {"value": ["team_1", "team_2", "team_3"], "test_column_1": [1, 2, 3]}
+    )
+    favorite_team = favorite_team.set_index("test_column_1")
+    favorite_color = pd.DataFrame(
+        {
+            "value": ["color_1", "color_2", "color_3"],
+            "test_column_2": ["value_1", "value_2", "value_3"],
+            "test_column_3_start": [0, 1, 2],
+            "test_column_3_end": [1, 2, 3],
+        }
+    )
+    favorite_color = favorite_color.set_index(
+        ["test_column_2", "test_column_3_start", "test_column_3_end"]
+    )
+    artifact_data = {
+        "simulants.favorite_team": favorite_team,
+        "simulants.favorite_color": favorite_color,
+    }
+    artifact = Artifact(hdf_file_path)
+    for key, data in artifact_data.items():
+        artifact.write(key, data)
+
+    component = ColumnCreator()
+    sim = InteractiveContext(components=[component], setup=False)
+    sim.configuration.update(
+        {
+            "input_data": {"artifact_path": hdf_file_path},
+            component.name: {
+                "lookup_tables": {
+                    "favorite_team": Component.build_lookup_table_config(
+                        value="data",
+                        categorical_columns=["test_column_1"],
+                        continuous_columns=[],
+                        key_name="simulants.favorite_team",
+                    ),
+                    "favorite_number": Component.build_lookup_table_config(
+                        value=0.4,
+                    ),
+                    "favorite_color": Component.build_lookup_table_config(
+                        value="data",
+                        categorical_columns=["test_column_2"],
+                        continuous_columns=["test_column_3"],
+                        key_name="simulants.favorite_color",
+                    ),
+                    "baking_time": Component.build_lookup_table_config(
+                        value="data",
+                        categorical_columns=["test_column_1", "test_column_2"],
+                        continuous_columns=["test_column_3"],
+                        key_name="simulants.baking_time",
+                        skip_build=True,
+                    ),
+                },
+            },
+        }
+    )
+    sim.setup()
+
+    # Assertions for specific lookup tables
+    assert component.lookup_tables["favorite_team"].key_columns == ["test_column_1"]
+    assert not component.lookup_tables["favorite_team"].parameter_columns
+    assert component.lookup_tables["favorite_color"].key_columns == ["test_column_2"]
+    assert component.lookup_tables["favorite_color"].parameter_columns == ["test_column_3"]
```

Comment:
I think you're missing a case where a lookup table built without key column but with parameter cols?